tests/test_csc.py

```diff
@@ -1,141 +1,69 @@
-                    continue  # valid command in DISABLED state
-                with self.subTest(bad_command=bad_command):
-                    cmd_attr = getattr(harness.remote, f"cmd_{bad_command}")
-                    with self.assertRaises(salobj.AckError):
-                        id_ack = await cmd_attr.start(cmd_attr.DataType(), timeout=SHORT_TIMEOUT)
-
-            # send enable; new state is ENABLED
-            cmd_attr = getattr(harness.remote, "cmd_enable")
-            harness.remote.evt_summaryState.flush()
-            id_ack = await cmd_attr.start(timeout=SHORT_TIMEOUT)
-            state = await harness.remote.evt_summaryState.next(flush=False, timeout=SHORT_TIMEOUT)
-            self.assertEqual(id_ack.ack, salobj.SalRetCode.CMD_COMPLETE)
-            self.assertEqual(id_ack.error, 0)
-            self.assertEqual(harness.csc.summary_state, salobj.State.ENABLED)
-            self.assertEqual(state.summaryState, salobj.State.ENABLED)
-
-            self.assertIsNotNone(harness.csc.telemetry_loop_task)
-            self.assertIsNotNone(harness.csc.seeing_loop_task)
-
-            for bad_command in commands:
-                if bad_command == "disable":
-                    continue  # valid command in ENABLE state
-                with self.subTest(bad_command=bad_command):
-                    cmd_attr = getattr(harness.remote, f"cmd_{bad_command}")
-                    with self.assertRaises(salobj.AckError):
-                        id_ack = await cmd_attr.start(timeout=SHORT_TIMEOUT)
-
-            # check that received telemetry topic from dimm
-            try:
-                await harness.remote.tel_status.next(
-                    flush=True, timeout=salobj.base_csc.HEARTBEAT_INTERVAL*5)
-            except asyncio.TimeoutError:
-                self.assertTrue(False, 'No status published by DIMM')
-
-            # check that received measurement from dimm
-            try:
-                await harness.remote.evt_dimmMeasurement.next(flush=False,
-                                                              timeout=10)
-            except asyncio.TimeoutError:
-                self.assertTrue(False, 'No measurement published by DIMM.')
-
-            # send disable; new state is DISABLED
-            cmd_attr = getattr(harness.remote, "cmd_disable")
-            # this CMD may take some time to complete
-            id_ack = await cmd_attr.start(timeout=60.)
-            self.assertEqual(id_ack.ack, salobj.SalRetCode.CMD_COMPLETE)
-            self.assertEqual(id_ack.error, 0)
-            self.assertEqual(harness.csc.summary_state, salobj.State.DISABLED)
-
-
-if __name__ == '__main__':
-    unittest.main()
+        logging.info("test_standard_state_transitions")
+        async with self.make_csc(
+            initial_state=salobj.State.STANDBY, config_dir=None, simulation_mode=1
+        ):
+            await self.check_standard_state_transitions(
+                enabled_commands=(
```

Comment:
I'm not sure you need this because they are generic commands.

Reply:
Good point. I added them because they were in the original test but they indeed are not needed so I removed them again.